clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -393,11 +392,9 @@ static void visitFunctionCallArguments(IndirectLocalPath &Path, Expr *Call,
     Path.push_back({IndirectLocalPathEntry::LifetimeBoundCall, Arg, D});
     if (Arg->isGLValue())
       visitLocalsRetainedByReferenceBinding(Path, Arg, RK_ReferenceBinding,
-                                            Visit,
-                                            /*EnableLifetimeWarnings=*/false);
+                                            Visit);
     else
-      visitLocalsRetainedByInitializer(Path, Arg, Visit, true,
-                                       /*EnableLifetimeWarnings=*/false);
```

Comment:
We change the behavior on this line and the line above. (Previously, we never run the GSL code path for the `Arg`).

The fix is on Line 409.

Reply:
I think the PR needs to be renamed. From a PR with the title `Remove the EnableLifetimeWarnings flag in lifetime analysis.` I'd expect either an NFC or removing a feature, not an actual bug fix. 

I think we should either split this up in two PRs, or rename the title to describe the bugfix aspect.

Reply:
Good point. It is hard to split in two PRs, refined the PR to focus on the bug fixes rather than the refactoring change.